src/sys/run_metrics.cxx

```diff
@@ -0,0 +1,73 @@
+/*!************************************************************************
+ * Copyright 2010 B.D.Dudson, S.Farley, M.V.Umansky, X.Q.Xu
+ *
+ * Contact Ben Dudson, bd512@york.ac.uk
+ * 
+ * This file is part of BOUT++.
+ *
+ * BOUT++ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * BOUT++ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with BOUT++.  If not, see <http://www.gnu.org/licenses/>.
+ * 
+ *********************************************************/
+
+#include <bout/run_metrics.hxx>
+#include <datafile.hxx>
+#include <output.hxx>
+
+/*!
+ * Adds variables to the output file, for post-processing
+ */
+void RunMetrics::outputVars(Datafile &file) {
+
+  file.add(t_elapsed, "wall_time", true);
+  file.add(wtime, "wtime", true);
+  file.add(ncalls, "ncalls", true);
+  file.add(ncalls_e, "ncalls_e", true);
+  file.add(ncalls_i, "ncalls_i", true);
+  file.add(wtime_rhs, "wtime_rhs", true);
+  file.add(wtime_invert, "wtime_invert", true);
+  file.add(wtime_comms, "wtime_comms", true);
+  file.add(wtime_io, "wtime_io", true);
+  file.add(wtime_per_rhs, "wtime_per_rhs", true);
+  file.add(wtime_per_rhs_e, "wtime_per_rhs_e", true);
+  file.add(wtime_per_rhs_i, "wtime_per_rhs_i", true);
+}
+
+void RunMetrics::calculateDerivedMetrics() {
+
+  wtime_per_rhs = wtime / ncalls;
+  wtime_per_rhs_e = wtime / ncalls_e;
```

Comment:
I'm not sure this and `wtime_per_rhs_i` will be useful as `wtime` is the total time for the output step, which will have some non-trivial split between the explicit and implicit parts. If `ncalls==ncalls_e+ncalls_i` in this case then the first metric is still partially useful. The second two might be more useful as `ncalls_e/ncalls` and `ncalls_i/ncalls` here so that they're fraction of calls that are explicit/implicit. @bendudson may be better positioned to comment here though as I've not worked with the split schemes for a while.